config/lfsc.m4

```diff
@@ -0,0 +1,87 @@
+# CVC4_CHECK_FOR_LFSC
+# ------------------
+# Look for LFSC and link it in, but only if user requested.
+AC_DEFUN([CVC4_CHECK_FOR_LFSC], [
+AC_MSG_CHECKING([whether user requested LFSC support])
+
+have_liblfsc=0
+LFSC_LIBS=
+LFSC_LDFLAGS=
+
+have_liblfsc=0
+if test "$with_lfsc" = no; then
+  AC_MSG_RESULT([no, LFSC disabled by user])
+elif test -n "$with_lfsc"; then
+  AC_MSG_RESULT([yes, LFSC requested by user])
+  AC_ARG_VAR(LFSC_HOME, [path to top level of LFSC source tree])
+  AC_ARG_WITH(
+    [lfsc-dir],
+    AS_HELP_STRING(
+      [--with-lfsc-dir=PATH],
+      [path to top level of lfsc source tree]
+    ),
+    [LFSC_HOME="$withval"],
+    []
+  )
+
+  if test -z "$LFSC_HOME" -a -e "$ac_abs_confdir/lfsc-checker"; then
+    AC_MSG_CHECKING([for LFSC checker library])
+    LFSC_HOME="$ac_abs_confdir/lfsc-checker/install"
+    AC_MSG_RESULT([found LFSC checker in $LFSC_HOME])
+  fi
+  
+  if test -z "$LFSC_HOME"; then
+    AC_MSG_FAILURE([must give --with-lfsc-dir=PATH or define environment variable LFSC_HOME!])
+  fi
+
+  if ! test -d "$LFSC_HOME" || ! test -x "$LFSC_HOME/bin/lfscc" ; then
+    AC_MSG_FAILURE([either $LFSC_HOME is not a LFSC install tree or it's not yet built])
+  fi
+
+  CPPFLAGS="$CPPFLAGS -I$LFSC_HOME/include"
+
+  AC_MSG_CHECKING([how to link LFSC])
+  CVC4_TRY_LFSC_LIB
+
+  if test -z "$LFSC_LIBS"; then
+    AC_MSG_FAILURE([cannot link against liblfscc!])
+  else
+    AC_MSG_RESULT([$LFSC_LIBS])
+    have_liblfsc=1
+  fi
+
+  LFSC_LDFLAGS="-L$LFSC_HOME/lib"
+
+else
+  AC_MSG_RESULT([no, user didn't request LFSC])
+  with_lfsc=no
+fi
+
+])# CVC4_CHECK_FOR_LFSC
+
+# CVC4_TRY_LFSC_LIB
+# ------------------------------
+# Try AC_CHECK_LIB(lfsc) with the given linking libraries
+AC_DEFUN([CVC4_TRY_LFSC_LIB], [
+if test -z "$LFSC_LIBS"; then
```

Comment:
Yes, in this case it's empty. I thought it might be a good idea to check if it's empty in case we want to add more CVC4_TRY_LFSC_LIB with different libs (similar to cryptominisat and abc). But I guess we can just remove the check.